### DISCUSSION TITLE
Help with TRV and radiator inside small encapsulated space - self regulation

### DISCUSSION THREAD

**Discussion by 0verEngineer** [jmcollin78/versatile_thermostat#648] [Q&A] [answered]
Hey, i have a TRV and a radiator inside a small encapsulated space the problem is the self regulation does not help very well in this case so i tried to use the expert mode and set the `offset_max` to 8 but this does not help me very much.

The problem is it does not heat most of the time because it is so warm where the TRV is and the room temp is too low most of the times and it takes a lot of time to heat the room because of this.

I don't quite understand the self regulation variables.

What of the variables should i modify to get it to heat probably?

**Comment by jmcollin78** [comment]
Hello, to help with regulation we need to have some curve. You have example of regulation curve in the readme (https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#algorithm).
We also need to know which Vtherm type you use and how it is configured. So please post the attributes of your Vtherm (https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#algorithm)

**Comment by 0verEngineer** [comment]
Sorry, i forgot to add it, here are the attributes:

```
hvac_modes: auto, heat, off
min_temp: 7
max_temp: 35
target_temp_step: 0.5
preset_modes: none, frost, eco, comfort, boost
current_temperature: 22.2
temperature: 23
hvac_action: heating
preset_mode: comfort
is_on: true
hvac_mode: heat
type: null
is_controlled_by_central_mode: true
last_central_mode: null
frost_temp: 24
eco_temp: 21
boost_temp: 25
comfort_temp: 23
frost_away_temp: 0
eco_away_temp: 0
boost_away_temp: 0
comfort_away_temp: 0
power_temp: null
target_temperature_step: 0.5
ext_current_temperature: 1.79
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: comfort
saved_target_temp: 23
saved_hvac_mode: null
motion_sensor_entity_id: null
motion_state: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
overpowering_state: null
presence_sensor_entity_id: null
presence_state: null
window_state: off
window_auto_state: off
window_bypass_state: false
window_sensor_entity_id: null
window_delay_sec: null
window_auto_enabled: false
window_auto_open_threshold: null
window_auto_close_threshold: null
window_auto_max_duration: null
window_action: window_turn_off
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-11-16T22:57:16.025148+01:00
last_ext_temperature_datetime: 2024-11-16T23:07:47.473103+01:00
security_state: false
minimal_activation_delay_sec: 10
device_power: 1
mean_cycle_power: null
total_energy: 246.57
last_update_datetime: 2024-11-16T23:07:47.495244+01:00
timezone: Europe/Berlin
temperature_unit: °C
is_device_active: true
ema_temp: 22.45
is_used_by_central_boiler: false
temperature_slope: -1.44
hvac_off_reason: null
is_over_climate: true
start_hvac_action_date: 2024-11-16T22:46:50.149637+01:00
underlying_entities: climate.thermostat_07_omi_esszimmer
is_regulated: true
regulated_target_temperature: 27.5
auto_regulation_mode: auto_regulation_strong
regulation_accumulated_error: 50
auto_fan_mode: auto_fan_none
current_auto_fan_mode: auto_fan_none
auto_activated_fan_mode: null
auto_deactivated_fan_mode: null
auto_regulation_use_device_temp: false
auto_start_stop_enable: false
auto_start_stop_level: auto_start_stop_none
auto_start_stop_dtmin: null
auto_start_stop_accumulated_error: 0
auto_start_stop_accumulated_error_threshold: null
follow_underlying_temp_change: false
friendly_name: versatile_thermostat_omi_esszimmer
supported_features: 401
```


and here is the graph:

![image](https://github.com/user-attachments/assets/b00646b7-acea-408a-881e-b3de8be8dd71)

**Comment by 0verEngineer** [comment]
The radiators don't work over night so thats why the temp falls over night

**Comment by 0verEngineer** [comment]
This is my expert mode config:

```
versatile_thermostat:
  auto_regulation_expert:
    kp: 0.4
    ki: 0.08
    k_ext: 0.0
    offset_max: 8
    stabilization_threshold: 0.1
    accumulated_error_threshold: 50
```

**Comment by jmcollin78** [comment]
Hello have you tried to remove all self-regulation? Your environment is very reactive and don't need self-regulation I guess.

**Comment by 0verEngineer** [comment]
Ok i will try, will it still modify the target temp of the TRV? And what is the maximum the target temp will be modified?

**Comment by 0verEngineer** [comment]
After the 15 minutes that i have self-regulation turned off the target temp of the TRV is still 23, exactly like the target temp of the VT, the local temp of the thermostat is 28 and the room temp of the external sensor is 22.3

So it should heat but the target temp of the TRV is not changed so that it heats.

**Reply by jmcollin78** [reply]
Ok that is better not not totally. Tuning and find the right option is always a mess.

What you can try now is to check the "use internal device temperature". Then VTherm will try to compensate the difference between internal temp of the device and the real temp of the room.

**Reply by 0verEngineer** [reply]
Ok should i use this option with self-regulation disabled?

**Reply by jmcollin78** [reply]
Yes

**Reply by 0verEngineer** [reply]
It sets the target temp of the TRV to the internal temp of the TRV + 0.5, (if the room temp is 22.3 and the target temp 23) so this does not help me because of following reasons:
- the thermostat does not open at all when the target temp is only 0.5 higher then the room temp
- the thermostat does only open a little bit when the target temp is 1 higher then the room temp but i need it fully open to heat the room to the target

is there a way to tune this `use internal device temperature` setting so that it works for me?

**Reply by jmcollin78** [reply]
Have you some curves yet ?

**Reply by 0verEngineer** [reply]
No they are not clean because i played with the settings, but my description should be clear enough (and more important then a graph).

This setting leads to the problem that the TRVs target temperature is 0.5 higher then the TRVs local temp, so it does not heat when it should heat.

Is there some way to tune this setting to have a higher target temp?

**Comment by jmcollin78** [comment]
> Is there some way to tune this setting to have a higher target temp?

Are you French ? I think we have a big misunderstanding.

**Reply by 0verEngineer** [reply]
No, i am german, can you somehow explain what you don't understand ir where the misunderstanding is?

**Reply by jmcollin78** [reply]
I will try to synthetize what I understand of your issue:

<img width="255" alt="Capture d’écran 2024-11-18 à 22 14 00" src="https://github.com/user-attachments/assets/349bea28-77bd-43fe-8773-498ab5b3145a">

- your have an `over_climate` VTherm with Strong regulation,
- your VTherm is always over heating. At each moment of the day the regulated temperature is over the target. This is the first very strange thing: the regulated should be under the target when it is over heating,
- you remove totally the self-regulation and the behavior is the same. This is totally un-understandable,
- you have try with an expert regulation parameters and nothing change. I have no curves to verify this. This is the 2nd very strange behavior. You should definitively remarks some changes,
- you have try with 'use the internal temperature' and nothing change (I still don't have any curves to verify). This is even more strange.
- when i look at your attributes, it seems I see:
```
current_temperature: 22.2
temperature: 23
```
This is not normal what I see in the curves. Maybe this is a clue to understand what happens.

If I correctly understand your case, it seems your VTherm don't work at all or I totally misunderstood your case.

**Reply by 0verEngineer** [reply]
Ok i think your misunderstanding comes from the graph, don't focus too much on the graph in my case because as i have written at the beginning, this TRV and the radiator/heater is in a **small encapsulated space** so it gets very hot very fast in there.

- I tried to compensate this problem with the self-regulation first but this did not help
- Then i tried your suggestion with the `use internal temperature` checkbox but this did also not hell


The problem is the TRV does not heat very good because it is so hot in the soace where it is but the room is still cold.

**Reply by jmcollin78** [reply]
> don't focus too much on the graph in my case

The only way to find what goes wrong, is to look at the graph. That is why I ask for the graph.

> the room is still cold.

😮‍💨 This is not what I see. Current temp (orange) is way above the target (yellow). That is why I try to lower the heating.

Don't tell me that the correct temperature is the purple and not the orange...
<img width="341" alt="Capture d’écran 2024-11-18 à 22 41 21" src="https://github.com/user-attachments/assets/301ae6df-1869-4068-ba27-32221b8cba1e">

In the first post:
> if the time because it is so warm where the TRV is

This makes me understand that it was too warm.

**Reply by jmcollin78** [reply]
I just need that : https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#even-better-with-plotly-to-tune-your-thermostat
not less and no more.

**Reply by 0verEngineer** [reply]
Yea the purple one is the room temp 😅
Sorry, i should have made this clear..

**Reply by 0verEngineer** [reply]
I have set it to 7 days here because the values of today are bad because i played with a lot of things
![image](https://github.com/user-attachments/assets/719d5633-58e8-4f5c-9631-7f8383f0a443)

**Reply by jmcollin78** [reply]
ok that is clear now. Even with a strong regulation (regulation t° >>> target), the heater don't heat.
So what you need is an stronger regulation. My fear is that you risk to exceed the max allowed `max_temp: 35` and then it will do nothing more. Let's try anyway.

So:
1. put the **Regulation level to 'Expert'**
2. change the expert parameters (in your configuration.yaml) like that:
```
    kp: float = 0.6
    ki: float = 0.1
    k_ext: float = 0.2
    offset_max: float = 10
    stabilization_threshold: float = 0.1
    accumulated_error_threshold: float = 80
```
3. un check the 'use internal device temperature'

When you change the configuration.yaml don't forget to restart.

This should be too much, but if it works we will lower the values after.

**Reply by 0verEngineer** [reply]
It is a little bit better but still not good :/

1Day:
![image](https://github.com/user-attachments/assets/2c2c5d01-1785-43c2-a086-df524bf76046)

**Reply by jmcollin78** [reply]
offset_max: float = 15. try to adjust this value

**Reply by 0verEngineer** [reply, marked as answer]
I tested it since you suggested it and it seems to work fine.
